src/popper/runner_host.py

```diff
@@ -293,6 +293,198 @@ def _update_with_engine_config(self, container_args):
+        args = {
+            "image": img,
+            "command": list(step.args),
+            "name": name,
+            "volumes": [f"{self._config.workspace_dir}:/workspace",],
+            "working_dir": step.dir if step.dir else "/workspace",
+            "environment": self._prepare_environment(step),
+            "entrypoint": step.runs if step.runs else None,
+            "detach": not self._config.pty,
+            "tty": self._config.pty,
+            "stdin_open": self._config.pty,
+        }
+
+        self._update_with_engine_config(args)
+
+        log.debug(f"container args: {pu.prettystr(args)}\n")
+
+        return args
+
+    def _update_with_engine_config(self, container_args):
+        """Given container arguments, it extends it so it includes options
+        obtained from the popper.config.Config.engine_opts property.
+        """
+        update_with = self._config.engine_opts
+        if not update_with:
+            return
+
+        container_args["volumes"] = [
+            *container_args["volumes"],
+            *update_with.get("volumes", list()),
+        ]
+        for k, v in update_with.get("environment", dict()).items():
+            container_args["environment"].update({k: v})
+
+        for k, v in update_with.items():
+            if k not in container_args.keys():
+                container_args[k] = update_with[k]
+
+    def _create_container(self, cid, step):
+        build, img, tag, build_ctx_path = self._get_build_info(step)
+
+        if build:
+            log.info(f"[{step.id}] podman build {img}:{tag} {build_ctx_path}")
+            if not self._config.dry_run:
+                pass
+        elif not self._config.skip_pull and not step.skip_pull:
+            log.info(f"[{step.id}] podman build {img}:{tag}")
+            if not self._config.dry_run:
+                cmd = ["podman", "pull", f"{img}:{tag}"]
+                HostRunner._exec_cmd(cmd, logging=False)
+
+        if self._config.dry_run:
+            return
+
+        container_args = self._get_container_kwargs(step, f"{img}:{tag}", cid)
+
+        log.debug(f"Container args: {container_args}")
+
+        msg = [f"{step.id}", "podman", "create", f"name={cid}"]
+        msg.append(f"image={container_args.get('image')}")
+        msg.append(f"entrypoint={container_args.get('entrypoint')}" or "")
+        msg.append(f"command={container_args.get('entrypoint')}" or "")
+        log.info(msg)
+
+        cmd = ["podman", "create"]
+        cmd.extend(["--name", container_args.get("name") or ""])
+        cmd.extend(["-v", container_args.get("volumes")[0] or ""])
+        if container_args.get("environment"):
+            for i, j in container_args["environment"].items():
+                cmd.extend(["-e", f"{i}={j}"])
+        cmd.append("-d" if container_args.get("detach") else "")
+        cmd.extend(["-w", container_args.get("working_dir") or ""])
+        cmd.extend(["-h", container_args.get("hostname") or ""])
```

Comment:
when `hostname` is not defined, this will only pass `-h`, which will cause the command to fail

Reply:
I corrected this! Argh! I didn't make the PR. it failed and I noticed this

Reply:
So how do you advise I pass this command using container.args?

Reply:
so this would be either empty or `-h <hostname>`, any way you find to express that is fine, for example:

```python
h = container_args.get("hostname", None)
if h:
  cmd.extend(f"-h {h}")
```

Reply:
Oh excellent, I was thinking of one-liners that's why I couldn't come up with this